fastestimator/trace/io/model_saver.py

```diff
@@ -12,10 +12,84 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ==============================================================================
+import os
+from typing import Union
 
-from fastestimator.trace.trace import Trace
+import numpy as np
+import tensorflow as tf
+import torch
+
+from fastestimator.trace import Trace
+from fastestimator.util import Data
 
 
 class ModelSaver(Trace):
-    def __init__(self, model):
-        pass
+    """Save trained model in hdf5/pt format.
+
+    Args:
+        model: Model instance defined by user (tf.keras.Model or torch.nn.Module)
+        model_name (str): Name of FE model.
+        save_dir (str): Directory to save the trained models.
+        save_freq (int, optional): Number of epochs to save models. Cannot be used with `save_best_only=True`. Defaults
+            to 1.
+        save_best (bool, str, optional): Best model saving monitor name. If True, the model loss is used. Defaults to
+            False.
+        save_best_mode (str, optional): Can be `'min'`, `'max'`, or `'auto'`. Defaults to 'max'.
+
+    """
+    def __init__(self, model: Union[tf.keras.Model, torch.nn.Module], model_name: str,
+                 save_dir: str, save_freq: int = 1, save_best: Union[bool, str]=False, save_best_mode:str='max'):
+
+        if save_best is True:
+            save_best = 'accuracy'
```

Comment:
why would we assume 'accuracy' instead of something like loss? perhaps don't allow boolean as an argument for save_best